### PR TITLE
JIT: Use metadata names for SIMD intrinsic method recognition

### PR DESCRIPTION
**Pull request by jakobbotsch**
eeGetMethodName is only expected to be used for diagnostics.

In practice this does not cause issues, but some odd situations could occur in the cases where `eeGetMethodName` returns a placeholder name during SPMI replay.

Also avoid unnecessary EE call in `impSIMDIntrinsic` in the common case of no SIMD intrinsic.

### CODE REVIEW

**Comment by jkotas** [comment]
If I am reading the code correctly, we are always going to ask for `getMethodNameFromMetadata` twice for SIMD methods. First time during attempt to import non-SIMD intrinsic and second time here. Would it be worth it to avoid these duplicate calls?

**Comment by jkotas** [comment]
If it is too complex refactoring, we should `if (isIntrinsic)` condition around the following in importer.cpp:
```
    {
        call = impSIMDIntrinsic(opcode, newobjThis, clsHnd, methHnd, sig, mflags, pResolvedToken->token);
        if (call != nullptr)
        {
            bIntrinsicImported = true;
            goto DONE_CALL;
        }
```
and flip the same check in `impSIMDIntrinsic` to assert. It will avoid the more expensive `isSIMDClass` checks for nearly all methods calls.

**Comment by jakobbotsch** [comment]
It's a good point. The refactoring does look a little complicated to do, but `impImportCall` is long overdue for a refactor so maybe I can spend some time on that. I also noticed that we needlessly query the signature for all hardware intrinsics even though we already have it from the call info.

**Comment by jakobbotsch** [comment]
@jkotas I started with a larger refactoring but as you mentioned it does end up being a more major operation, so for this PR I went with your second suggestion instead.

File: src/coreclr/vm/jitinterface.cpp [hunk 1]
```diff
@@ -6263,30 +6275,18 @@ const char* CEEInfo::getMethodNameFromMetadata(CORINFO_METHOD_HANDLE ftnHnd, con
...
+        {
+            IfFailThrow(pMDImport->GetNameOfTypeDef(pMT->GetCl(), className, namespaceName));
+        }
         // Query enclosingClassName when the method is in a nested class
         // and get the namespace of enclosing classes (nested class's namespace is empty)
-        if (pMT->GetClass()->IsNested())
+        if (enclosingClassName != NULL && pMT->GetClass()->IsNested())
         {
-            IfFailThrow(pMDImport->GetNameOfTypeDef(pMT->GetEnclosingCl(), &enclosingResult, &namespaceResult));
+            IfFailThrow(pMDImport->GetNameOfTypeDef(pMT->GetEnclosingCl(), enclosingClassName, namespaceName));
```
Comment (jakobbotsch) [inline src/coreclr/vm/jitinterface.cpp:6286]: This logic seems a bit odd to me, IIUC it will work only for one level of nesting.

Comment (jkotas) [reply src/coreclr/vm/jitinterface.cpp:6286]: This is designed to handle just enough for the JIT to identify intrinsics. We only need one level of nesting for that.

**Review by jkotas — APPROVED** [review]

**Comment by jakobbotsch** [comment]
Failure is known according to build analysis.

**Pull request merged**